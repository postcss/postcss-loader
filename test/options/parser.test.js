const { webpack } = require('@webpack-utilities/test')

describe('Options', () => {
  test('Parser - {String}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          parser: 'sugarss'
        }
      }
    }

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toEqual('module.exports = "a {\\n  color: black\\n}\\n"')
      expect(source).toMatchSnapshot()
    })
  })

  test('Parser - {Object}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          ident: 'postcss',
          parser: require('sugarss')
        }
      }
    }

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toEqual('module.exports = "a {\\n  color: black\\n}\\n"')
      expect(source).toMatchSnapshot()
    })
  })
})
