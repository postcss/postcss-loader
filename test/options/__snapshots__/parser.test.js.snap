// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Parser should emit error Parser: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
TypeError: parser is not a function",
  "ModuleError: Module Error (from \`replaced original path\`):
(Emitted value instead of an instance of Error) Loading PostCSS Parser failed: Cannot find module 'unresolve' from 'src/index.js'",
]
`;

exports[`Options Parser should emit error Parser: warnings 1`] = `Array []`;

exports[`Options Parser should work Parser - {Object}: css 1`] = `
"a {
  color: black
}
"
`;

exports[`Options Parser should work Parser - {Object}: errors 1`] = `Array []`;

exports[`Options Parser should work Parser - {Object}: warnings 1`] = `Array []`;

exports[`Options Parser should work Parser - {String}: css 1`] = `
"a {
  color: black
}
"
`;

exports[`Options Parser should work Parser - {String}: errors 1`] = `Array []`;

exports[`Options Parser should work Parser - {String}: warnings 1`] = `Array []`;
